epicsdbbuilder/const_array.py

```diff
@@ -0,0 +1,92 @@
+from decimal import Decimal
+from .recordbase import quote_string
+from .parameter import Parameter
+
+__all__ = ['ConstArray']
+
+
+class ConstArray:
+    """Constant Link Values. EPICS Base 7.0.4.1 and above.
+
+    Example: PY Source
+    ------------------
+    `r = records.lsi('r', INP=ConstArray(['Plain String not DBLINK']))`
+
+    Example: Generated DB
+    ---------------------
+    `field(INP, ["Plain String not DBLINK"])`
+    """
+
+    def __init__(self, value):
+        self.value = value
+
+    @property
+    def value(self):
+        return self._value
+
+    @value.setter
+    def value(self, value):
+        self._assert_valid(value)
+        self._value = value
+
+    def _assert_valid(self, value, record=None, fieldname=None):
+        id = 'ConstArray' \
```

Comment:
I am not sure I got the comment right. Yet, I am going to assume you are referring to the method `Validate`. In such a case, the method `_assert_valid' is used by the setter of a property `value`. There, I have no record and no fieldname. Definitely, I can move "None" there to avoid the default values here or avoid the check until the DB is generated. I just thought checking it early might be more convenient. What would you prefer?